diesel_derives/src/selectable.rs

```diff
@@ -28,9 +29,43 @@ pub fn derive(item: DeriveInput) -> TokenStream {
 
     let struct_name = &item.ident;
 
-    let field_columns_ty = model.fields().iter().map(|f| field_column_ty(f, &model));
+    let field_columns_ty = model
+        .fields()
+        .iter()
+        .map(|f| field_column_ty(f, &model))
+        .collect::<Vec<_>>();
     let field_columns_inst = model.fields().iter().map(|f| field_column_inst(f, &model));
 
+    let field_check_bound = model
+        .fields()
+        .iter()
+        .zip(&field_columns_ty)
+        .filter(|(f, _)| !f.embed())
+        .flat_map(|(f, ty)| {
+            let field_ty = to_field_ty_bound(&f.ty);
+            let span = field_ty.span();
+            let mut out = Vec::new();
+            if let Some(field_ty) = field_ty {
+
+                if cfg!(feature = "postgres") {
+                    out.push(quote::quote_spanned! {span=>
+                                                     #field_ty: diesel::prelude::Queryable<diesel::dsl::SqlTypeOf<#ty>, diesel::pg::Pg>
```

Comment:
Isn't the `Queryable` requirement a bit harsh? (since it's essentially a helper to implement `FromSqlRow` we should probably require that instead)